library/alloc/src/str.rs

```diff
@@ -383,15 +383,45 @@ impl str {
                   without modifying the original"]
     #[stable(feature = "unicode_case_mapping", since = "1.2.0")]
     pub fn to_lowercase(&self) -> String {
-        let mut s = String::with_capacity(self.len());
-        for (i, c) in self[..].char_indices() {
+        let mut out = Vec::<u8>::with_capacity(self.len());
+        let b = self.as_bytes();
+
+        const UNROLL: usize = 128;
```

Comment:
Could you add a comment explaining that this is a fast path for ASCII text, and that `UNROLL` is used to consume text in big chunks? It would help people reading the stdlib source.